libs/d3d12core/main.c

```diff
@@ -53,6 +54,197 @@ static void *vulkan_module = NULL;
 #endif
 static PFN_vkGetInstanceProcAddr vulkan_vkGetInstanceProcAddr = NULL;
 
+#ifdef _WIN32
+static BOOL wait_vr_key(HKEY vr_key)
+{
+    DWORD type, value, wait_status, size;
+    DWORD max_retry = 10; /* Each value query and each timeout counts as a try. */
+    LSTATUS status;
+    HANDLE event;
+
+    size = sizeof(value);
+    if ((status = RegQueryValueExA(vr_key, "state", NULL, &type, (BYTE *)&value, &size)))
+    {
+        ERR("OpenVR: could not query value, status %d", status);
+        return false;
+    }
+    if (type != REG_DWORD)
+    {
+        ERR("OpenVR: unexpected value type %d", type);
+        return false;
+    }
+
+    if (value)
+        return value == 1;
+
+    event = CreateEventA(NULL, FALSE, FALSE, NULL);
+    if (event == NULL)
+    {
+        ERR("Cannot create event");
+        return false;
+    }
+
+    while (max_retry)
+    {
+        if (RegNotifyChangeKeyValue(vr_key, FALSE, REG_NOTIFY_CHANGE_LAST_SET, event, TRUE))
+        {
+            ERR("Error registering registry change notification");
+            break;
+        }
+        size = sizeof(value);
+        if ((status = RegQueryValueExA(vr_key, "state", NULL, &type, (BYTE *)&value, &size)))
+        {
+            ERR("OpenVR: could not query value, status %d", status);
+            break;
+        }
+        if (value)
+            break;
+        max_retry--;
+
+        while ((wait_status = WaitForSingleObject(event, 1000)) == WAIT_TIMEOUT && max_retry)
+        {
+            WARN("VR state wait timeout (retries left %u)", max_retry);
+            max_retry--;
```

Comment:
Need to continue; after this line, otherwise a misleading ERR is reported.

Reply:
I think it's better to exclude `WAIT_TIMEOUT` from the it below, if that what you meant?

Reply:
Or that. It should not print error on timeout.